app/assets/javascripts/discourse/app/components/reviewable-refresh/insights.gjs

```diff
@@ -0,0 +1,95 @@
+import Component from "@glimmer/component";
+import icon from "discourse/helpers/d-icon";
+import { i18n } from "discourse-i18n";
+
+/**
+ * Displays contextual insights about a reviewable item, including AI confidence,
+ * flag volume, user history, and moderation actions. Only shows relevant insights
+ * based on available data.
+ *
+ * @component ReviewableInsights
+ *
+ * @param {Reviewable} reviewable - The reviewable object containing all review data
+ */
+export default class ReviewableInsights extends Component {
+  /**
+   * Returns array of review insight objects to display
+   * @returns {Array<Object>} Array of insight configuration objects
+   */
+  get reviewInsights() {
+    const insights = [];
+    const reviewable = this.args.reviewable;
+    const user = this.args.reviewable.target_created_by;
+
+    // Flag volume insight
+    insights.push({
+      icon: "triangle-exclamation",
+      label: i18n("review.insights.flag_volume"),
+      description: i18n("review.insights.flagged_by_users", {
+        count: reviewable?.reviewable_scores?.length || 0,
+      }),
+    });
+
+    // Similar posts insight
+    if (user.flags_agreed) {
+      insights.push({
+        icon: "clock-rotate-left",
+        label: i18n("review.insights.similar_posts"),
+        description: i18n("review.insights.flagged_in_timeframe", {
+          count: user.user_stat.flags_agreed,
```

Comment:
The first issue I spotted is that my serialised users do not always have `user_stat`

<img width="2912" alt="Screenshot 2025-07-02 at 4 02 35 pm" src="https://github.com/user-attachments/assets/b603977e-68bd-492a-85be-c9ae871ac065" />

However, there is always access to `flags_agreed`

<img width="1035" alt="Screenshot 2025-07-02 at 4 03 03 pm" src="https://github.com/user-attachments/assets/aa976ba5-d332-4c14-b005-20a67a60878a" />


So `user.flags_agreed` work for me